x-pack/plugins/security_solution/public/cloud_security_posture/components/csp_details/insights_tab_csp.tsx

```diff
@@ -5,19 +5,105 @@
  * 2.0.
  */
 
-import React, { memo } from 'react';
-import { EuiSpacer } from '@elastic/eui';
+import React, { memo, useMemo, useState } from 'react';
+import type { EuiButtonGroupOptionProps } from '@elastic/eui';
+import { EuiButtonGroup, EuiSpacer } from '@elastic/eui';
+import { FormattedMessage } from '@kbn/i18n-react';
+import type { FlyoutPanelProps, PanelPath } from '@kbn/expandable-flyout';
+import { useExpandableFlyoutState } from '@kbn/expandable-flyout';
+import { i18n } from '@kbn/i18n';
+import type { FlyoutPanels } from '@kbn/expandable-flyout/src/store/state';
 import { MisconfigurationFindingsDetailsTable } from './misconfiguration_findings_details_table';
+import { VulnerabilitiesFindingsDetailsTable } from './vulnerabilities_findings_details_table';
 
 /**
  * Insights view displayed in the document details expandable flyout left section
  */
+
+interface CspFlyoutPanelProps extends FlyoutPanelProps {
+  params: {
+    path: PanelPath;
+    hasMisconfigurationFindings: boolean;
+    hasVulnerabilitiesFindings: boolean;
+  };
+}
+interface CspFlyoutPanels extends FlyoutPanels {
+  left: CspFlyoutPanelProps;
+}
+
 export const InsightsTabCsp = memo(
   ({ name, fieldName }: { name: string; fieldName: 'host.name' | 'user.name' }) => {
+    const panels = useExpandableFlyoutState() as CspFlyoutPanels;
+    const defaultTab = () =>
+      panels.left?.params?.hasMisconfigurationFindings
+        ? 'misconfigurationTabId'
+        : panels.left?.params?.hasVulnerabilitiesFindings
+        ? 'vulnerabilitiesTabId'
+        : 'undefined';
+
+    const [activeInsightsId, setActiveInsightsId] = useState(
+      panels.left?.params?.path?.subTab || defaultTab
+    );
+    const insightsButtonsNew: EuiButtonGroupOptionProps[] = useMemo(() => {
```

Comment:
nit, subjective and general comment. try to use spaces between constants. its much easier to read:

```ts
const user = {
  firstName: 'John',
  lastName: 'Doe',
  age: 30
};

const isActive = true;

const getFullName = () => {
  return `${user.firstName} ${user.lastName}`;
};

const address = {
  street: '123 Main St',
  city: 'Anytown'
};
```

compared to:

```ts
const user = {
  firstName: 'John',
  lastName: 'Doe',
  age: 30
};
const isActive = true;
const getFullName = () => {
  return `${user.firstName} ${user.lastName}`;
};
const address = {
  street: '123 Main St',
  city: 'Anytown'
};

```

no space between lines is better for multiple one liners like

```ts
const firstName = 'John';
const lastName = 'Doe';
const age = 30;
const isActive = true;
const address = '123 Main St';
const city = 'Anytown';
```

but it gets harder to read and review when there are multilines constants/functions and multiple indentations